### PR TITLE
[ios] add new Apple devices to constants

### PR DESCRIPTION
**Pull request by tsapeta**
It adds new Apple devices to `expo-constants` module. I've also refactored some stuff because these `if`s looked really awful 😂

Fixes https://github.com/expo/universe/issues/3150

### CODE REVIEW

**Review by sjchmiela — APPROVED** [review]
LGTM, just some nitpicks

File: packages/expo-constants/ios/EXConstants/EXConstantsService.m [hunk 1]
```diff
@@ -121,229 +122,289 @@ + (NSString *)devicePlatform
...
+                // Simulator
+                @"i386": @"Simulator",
+                @"x86_64": @"Simulator",
+                };
+  }
+
+  NSString *deviceModel = [mapping objectForKey:platform];
+
+  if (!deviceModel) {
+    // Not found on database. At least guess main device type from string contents:
```
[the remark this answers is not in the document]
Comment (tsapeta) [reply packages/expo-constants/ios/EXConstants/EXConstantsService.m]: that's right, it was copied from `react-native-device-info` 😂

**Comment by esamelson** [comment]
Thanks @tsapeta ! I am just wondering if initializing these large objects is going to add a chunk of time to TTI?

**Comment by tsapeta** [comment]
@esamelson: I've just ran some benchmarks and looks like initializing this dictionary is a bit slower than using so many `if`s but the difference is not big:

`if`:
```
0.056982ms
0.043988ms
0.037909ms
0.033975ms
0.033021ms
```

`dictionary`:
```
0.047088ms
0.043988ms
0.056028ms
0.057936ms
0.047922ms
```

What is interesting is that making the dictionary `static` increased these times significantly, to about ~0.25ms 😮

File: packages/expo-constants/ios/EXConstants/EXConstantsService.m [hunk 2]
```diff
@@ -124,226 +125,263 @@ + (NSString *)devicePlatform
...
+                            @"iPad4,5": @2013,
+                            @"iPad4,6": @2013,
+                            @"iPad4,7": @2014,
+                            @"iPad4,8": @2014,
+                            @"iPad4,9": @2014,
+                            @"iPad5,1": @2015,
+                            @"iPad5,2": @2015,
+                            };
+
+  NSNumber *deviceYear = [mapping objectForKey:platform];
```
Comment (ide) [inline packages/expo-constants/ios/EXConstants/EXConstantsService.m]:
```suggestion
  NSNumber *deviceYear = mapping[platform];
```

**Review by ide — APPROVED** [review]

**Comment by esamelson** [comment]
good to have those benchmarks to know this change doesn't cause an adverse effect. Thanks @tsapeta !

**Pull request merged**